apps/web/src/components/Shared/UserPreview.tsx

```diff
@@ -179,26 +176,30 @@ const UserPreview: FC<UserPreviewProps> = ({
     );
   };
 
-  const onPreviewStart = async () => {
-    if (!lazyProfile.id) {
-      setLoading(true);
-      const { data } = await loadProfile({
-        variables: {
-          request: { handle: formatHandle(lazyProfile?.handle, true) }
-        }
-      });
+  const Preview = () => {
+    if (loading) {
+      return (
+        <div className="flex flex-col">
+          <div className="horizontal-loader w-full">
+            <div />
+          </div>
+          <div className="flex p-3">
+            <div>{profile?.handle}</div>
+          </div>
+        </div>
+      );
+    }
 
-      const getProfile = data?.profile;
+    return <UserProfileContent compositeProfile={lazyProfile || profile} />;
+  };
 
-      if (getProfile) {
-        setLazyProfile(getProfile as Profile);
+  const onPreviewStart = () =>
+    !data &&
+    loadProfile({
+      variables: {
+        request: { handle: formatHandle(profile?.handle, true) }
       }
-
-      setTimeout(() => {
-        setLoading(false);
-      }, 500);
```

Comment:
we added 500ms delay intentionally to not load profile instantly 🙇🏼 can we re add back?

Reply:
[Added in last commit.](https://github.com/heyxyz/hey/pull/3884/commits/d09545c636ff38cee9de9798cee6bae58b97d959)